hazelcast-client/src/main/java/com/hazelcast/client/config/XmlClientConfigBuilder.java

```diff
@@ -324,11 +328,65 @@ private void handleNetwork(Node node) {
                 handleSSLConfig(child, clientNetworkConfig);
             } else if ("aws".equals(nodeName)) {
                 handleAWS(child, clientNetworkConfig);
+            } else if ("discovery-strategies".equals(nodeName)) {
+                handleDiscoveryStrategies(child, clientNetworkConfig);
             }
         }
         clientConfig.setNetworkConfig(clientNetworkConfig);
     }
 
+    private void handleDiscoveryStrategies(Node node, ClientNetworkConfig clientNetworkConfig) {
+        final DiscoveryStrategiesConfig discoveryStrategiesConfig = clientNetworkConfig.getDiscoveryStrategiesConfig();
+        for (Node child : new IterableNodeList(node.getChildNodes())) {
+            final String name = cleanNodeName(child.getNodeName());
+            if ("discovery-strategy".equals(name)) {
+                handleDiscoveryStrategy(child, discoveryStrategiesConfig);
+            } else if ("node-filter".equals(name)) {
+                handleDiscoveryNodeFilter(child, discoveryStrategiesConfig);
+            }
+        }
+    }
+
+    private void handleDiscoveryNodeFilter(Node node, DiscoveryStrategiesConfig discoveryStrategiesConfig) {
+        final NamedNodeMap atts = node.getAttributes();
+
+        final Node att = atts.getNamedItem("class");
```

Comment:
Can you remove final from local variables (unless there is an anonymous inner class that relies on it). Final variables only cause noise and we are getting rid of them throughout the system.